File: 1500-count-largest-group/1500-count-largest-group.java
class Solution {
    static int sum_of_digits(int num) {
        if (num == 0)
            return 0;
        return num % 10 + sum_of_digits(num / 10);
    }

    public int countLargestGroup(int n) {
        var sums = new int[37];

        for (int i = 1; i <= n; i++)
            sums[sum_of_digits(i)]++;
        int max = 0, cnt = 0;
        
        for (int i = 0; i < sums.length; i++) {
            if (sums[i] > max) {
                max = sums[i];
                cnt = 1;
            } else if (sums[i] == max)
                cnt++;
        }
        return cnt;
    }
}

File: 1500-count-largest-group/1500-count-largest-group.c
int countLargestGroup(int n) {
    int sums[37] = {};

    int sumOfDig(int num) {
        return (num == 0 ? 0 : (num % 10) + sumOfDig(num / 10));
    }

    for (int i = 1; i <= n; ++i)
        sums[sumOfDig(i)]++;

    int cnt = 0, max = 0;

    for (int i = 0; i < 37; i++)
        ((sums[i] > max) ? (max = sums[i], cnt = 1) : (sums[i] == max) ? ++cnt : 0);
        
    return cnt;
}